ui/candidate-portal/src/app/components/profile/view/tab/opps/opp/candidate-opp/candidate-opp.component.ts

```diff
@@ -50,24 +55,28 @@ export class CandidateOppComponent implements OnInit, OnChanges {
   }
 
   private fetchJobChats() {
+    const sourceChatRequest: CreateChatRequest = {
+      type: JobChatType.CandidateProspect,
+      candidateId: this.candidate.id
+    }
+    // Only want to show destination chat if candidate is at or further than the CV Review stage.
+    this.showDestinationChat = isOppStageGreaterThanOrEqualTo(this.selectedOpp?.stage, "cvReview")
     const destinationChatRequest: CreateChatRequest = {
       type: JobChatType.CandidateRecruiting,
       candidateId: this.candidate.id,
       jobId: this.selectedOpp?.jobOpp?.id
     }
-    const sourceChatRequest: CreateChatRequest = {
-      type: JobChatType.CandidateProspect,
-      candidateId: this.candidate.id
-    }
+    // Only want to show all job candidates chat if candidate is at or further than the Offer stage.
+    this.showAllChat = isOppStageGreaterThanOrEqualTo(this.selectedOpp?.stage, "offer");
     const allJobCandidatesChatRequest: CreateChatRequest = {
       type: JobChatType.AllJobCandidates,
       jobId: this.selectedOpp?.jobOpp?.id
     }
 
     forkJoin( {
       'sourceChat': this.chatService.getOrCreate(sourceChatRequest),
-      'destinationChat': this.chatService.getOrCreate(destinationChatRequest),
-      'allJobCandidatesChat': this.chatService.getOrCreate(allJobCandidatesChatRequest),
+      'destinationChat': this.showDestinationChat ? this.chatService.getOrCreate(destinationChatRequest) : of(null),
+      'allJobCandidatesChat': this.showAllChat? this.chatService.getOrCreate(allJobCandidatesChatRequest) : of(null),
     }).subscribe(
```

Comment:
Better to build the array of requests you want to forkJoin - and only load the ones you need.